[IO] Fix rootcp --replace flag not working with recursive copy

ahmedbilal9 opened the pull request:
Fixes #7052

## Changes or fixes:

The --replace flag was not being checked when copying regular objects (non-tree, non-collection) in recursive mode. This caused objects to be written with new cycles instead of replacing existing ones.

Added replaceOption check in the else block of copyRootObjectRecursive() to delete existing objects before writing replacements.

## Checklist:

- [ ] Tested changes locally
- [ ] Updated the docs (if necessary)

guitargeek commented:
Thank you very much for your PR! Could you fix the Python syntax errors? Thanks.

silverweed commented:
@ahmedbilal9 your latest commit does not compile. Please test all your changes locally before pushing them.

ferdymercury commented on roottest/main/CMakeLists.txt:
```diff
@@ -368,12 +368,27 @@ ROOTTEST_ADD_TEST(SimpleRootcp5CheckOutput
+ROOTTEST_ADD_TEST(RootcpReplaceRecursiveClean
+                  COMMAND ${PY_TOOLS_PREFIX}/rootrm${pyext} -r copy_replace_recursive.root
+                  FIXTURES_REQUIRED main-RootcpReplaceRecursiveCheckOutput-fixture)
 
```
```suggestion

#########################################################################

############################# ROOTMV TESTS ############################
ROOTTEST_ADD_TEST(SimpleRootmv1PrepareInput
                  COMMAND ${PY_TOOLS_PREFIX}/rootcp${pyext} --recreate -r test.root move1.root
                  FIXTURES_SETUP main-SimpleRootmv1PrepareInput-fixture)

```

ahmedbilal9 commented:
Hi @ferdymercury,

I've fixed the duplicate test issue in commit a65caee. The SimpleRootmv1PrepareInput test is now only at line 396 as you suggested, and I've removed the duplicate that was at line 433-438.

Sorry for the confusion earlier - I should have been more careful when manually applying your suggestion. I've verified the structure is correct now.

What should I do next to move this PR forward?

vepadulano commented on main/python/cmdLineUtils.py:
```diff
@@ -692,19 +692,18 @@ def copyRootObjectRecursive(sourceFile, sourcePathSplit, destFile, destPathSplit
+                retcodeTemp = deleteObject(destFile, destPathSplit + [actualName])
                 if retcodeTemp:
                     retcode += retcodeTemp
+                    obj.Delete()
```
Something not super clear to me: we call `deleteObject`, then we also call `obj.Delete` conditionally, why the apparently double delete?

ahmedbilal9 commented on roottest/main/CMakeLists.txt:
```diff
@@ -373,9 +368,31 @@ ROOTTEST_ADD_TEST(SimpleRootcp5CheckOutput
+
+ROOTTEST_ADD_TEST(RootcpReplaceRecursive
+                  COMMAND ${PY_TOOLS_PREFIX}/rootcp${pyext} --replace -r copy_replace_recursive.root:tof copy_replace_recursive.root:tof
+                  FIXTURES_REQUIRED main-RootcpReplaceRecursivePrepareInput-fixture
```
This test verifies that the --replace flag works correctly for in-place recursive copies. It copies from copy_replace_recursive.root:tof to the same location to ensure objects are replaced without creating new cycles (;2, ;3, etc.). The RootcpReplaceRecursiveCheckOutput test then verifies only cycle ;1 exists for each object, confirming replacement worked as intended.
